doc/develop/application/index.rst

```diff
@@ -44,25 +53,40 @@ These contents are:
 * **CMakeLists.txt**: This file tells the build system where to find the other
   application files, and links the application directory with Zephyr's CMake
   build system. This link provides features supported by Zephyr's build system,
-  such as board-specific kernel configuration files, the ability to run and
+  such as board-specific configuration files, the ability to run and
   debug compiled binaries on real or emulated hardware, and more.
 
-* **Kernel configuration files**: An application typically provides a
-  Kconfig configuration file (usually called :file:`prj.conf`) that specifies
-  application-specific values for one or more kernel configuration options.
-  These application settings are merged with board-specific settings to produce
-  a kernel configuration.
+* **app.overlay**: This is a devicetree overlay file that specifies
+  application-specific changes which should be applied to the base devicetree
+  for the board you are building for. The purpose of devicetree overlays is
```

Comment:
Maybe this should say "any board you are building for"?

I also wonder if we should provide guidance on board overlays here, since using board specific overlays allows your
application to be more portable between boards in the future. One pattern I expect customers use is to first get their sample app working on an EVK, then on a custom board. Board overlays would make this process a bit easier than `app.overlay` will.

Reply:
> I also wonder if we should provide guidance on board overlays here

I think not. The same question would apply to board-specific Kconfig fragments. I think both of these are a little too detailed for this overview, and they are already documented in the cross-references linked to from the overview.